### PR TITLE
Feature/wallet settings screen

### LINKED ISSUE #12: Add menu to wallet screen
Tapping the menu button on the wallet screen returns to the main screen.

Instead it should show a menu (side menu would be OK) with the following options:

- Show mnemonic:
```
const {read} = useKeychain();
const {wallet, walletName} = useWallet();

read(walletName).then((password: string) => {
const mnemonic: string = await wallet.db.GetMasterKey(
      "mnemonic",
      password
    );
});
```
- Configure use of biometrics after locking the app
```
  const [lockAfterBackground, setLockAfterBackground] = useAsyncStorage(
    'lockAfterBackground',
    'false',
  );
```
- Set up servers to connect. Should allow to set a list of nodes for both `testnet` and `mainnet`, stored locally. Creates an object with the following structure:
```
testnet: [
    {
      host: "electrum-testnet2.nav.community",
      port: 40004,
      proto: "wss",
    },
    {
      host: "electrum-testnet.nav.community",
      port: 40004,
      proto: "wss",
    },
  ],
  mainnet: [
    {
      host: "electrum4.nav.community",
      port: 40004,
      proto: "wss",
    },
    {
      host: "electrum.nextwallet.org",
      port: 40004,
      proto: "wss",
    },
    {
      host: "electrum2.nav.community",
      port: 40004,
      proto: "wss",
    },
    {
      host: "electrum3.nav.community",
      port: 40004,
      proto: "wss",
    },
    {
      host: "electrum.nav.community",
      port: 40004,
      proto: "wss",
    },
  ]
```
which is then passed with the `nodes` property of `WalletFile.Load()` in the `WalletProvider`. 
```
      walletFile.Load({
        bootstrap: njs.wallet.xNavBootstrap,
        nodes
      });
```
If nodes == undefined, then it will use the default list of nodes.
- Remove wallet: accomplished calling `njs.wallet.WalletFile.RemoveWallet(name)`, needs close the wallet first.
- Close wallet and back to main screen

### PR DESCRIPTION
**Pull request by tsejerome**
Resolves #12

### CODE REVIEW

**Comment by aguycalled** [comment]
image

Parsing of hosts fails

**Comment by aguycalled** [comment]
image

Protocol is surrounded by a `<b>` tag

**Comment by aguycalled** [comment]
I am thinking it would be useful to include a button to `Restore default servers` which would set the servers to those included in `constants/Data.ts`

**Comment by aguycalled** [comment]
Seeing this:

image

File: utils/server.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const validateIp = (host: string) => {
```
Comment (aguycalled) [inline utils/server.tsx:1]: Here it should use `is-valid-hostname`

**Comment by aguycalled** [comment]
The title of the Biometrics dialogs still shows the setting is active even if it's not.

image

**Comment by aguycalled** [comment]
Removing a wallet does not remove it.

**Comment by tsejerome** [comment]
> Seeing this:
> 
> image

https://stackoverflow.com/questions/60954742/how-to-pass-parent-function-to-child-screen-in-react-navigation-5
This is an intentional warning, and as we currently don't need state persistence or deep link in this page, we can ignore this for now. Any thots @aguycalled ?

**Comment by tsejerome** [comment]
✅ Validate hostname
✅ Remove b tag (was testing out adding b tag for certain words but failed)
✅ Restore default server button
✅ Fix Remove Wallet 

**Remaining:**
Fix on Biometrics

**Comment by tsejerome** [comment]
✅  Fix on Biometrics

**Comment by tsejerome** [comment]
Ready for review again!

**Comment by aguycalled** [comment]
Everything looking good now.

With `navcoin-js >=1.1.63`, the `Delete Wallet` and `Close Wallet` actions can be safely performed even if the wallet is not connected yet.

Will approve and merge once `navcoin-js` is upgraded and `Delete Wallet` and `Close Wallet` buttons are always shown.

**Pull request merged**